fabric-rendering-fluids-v1/src/main/java/net/fabricmc/fabric/api/client/render/fluid/v1/SimpleFluidRenderHandler.java

```diff
@@ -137,6 +138,59 @@ public static SimpleFluidRenderHandler coloredWater(int tint) {
 		return new SimpleFluidRenderHandler(WATER_STILL, WATER_FLOWING, WATER_OVERLAY, tint);
 	}
 
+	/**
+	 * @return The current texture for the fluid in still state.
+	 */
+	public Identifier getStillTexture() {
+		return textures[0];
+	}
+
+	/**
+	 * @return The current texture for the fluid in flowing state.
+	 */
+	public Identifier getFlowingTexture() {
+		return textures[1];
+	}
+
+	/**
+	 * @return The current texture for the fluid overlay.
+	 */
+	public Identifier getOverlayTexture() {
+		return textures[2];
+	}
+
+	/**
+	 * Set the current texture for the fluid in still state.
+	 * @param id Identifier of the texture.
+	 */
+	public void setStillTexture(Identifier id) {
+		textures[0] = id;
+	}
+
+	/**
+	 * Set the current texture for the fluid in flowing state.
+	 * @param id Identifier of the texture.
+	 */
+	public void setFlowingTexture(Identifier id) {
+		textures[1] = id;
+	}
+
+	/**
+	 * Set the current texture for the fluid overlay.
+	 * @param id Identifier of the texture.
+	 */
+	public void setOverlayTexture(Identifier id) {
+		textures[2] = id;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public Identifier[] getTexturesIds() {
+		return Arrays.stream(textures).filter(Objects::nonNull).toArray(Identifier[]::new);
```

Comment:
I would do this once in the constructor. Dont really want to be using a stream everytime.

Reply:
Fixed